### DISCUSSION TITLE
How to use sessionStorage as the infinite hits cache?

### DISCUSSION THREAD

**Discussion by sbbu** [algolia/instantsearch#5923] [Q&A]
https://www.algolia.com/doc-beta/ui-libraries/instantsearch/results/infinite-hits/react#cache

>The widget caches all loaded hits. By default, it uses its own internal in-memory cache implementation. Alternatively, use [sessionStorage](https://developer.mozilla.org/en-US/docs/Web/API/Window/sessionStorage) to retain the cache even if users reload the page.

How? The docs give no further details on how to use this. I have found some other examples which all seem to be broken or no longer supported

**Comment by Haroenv** [comment]
The regular docs has an example: https://www.algolia.com/doc/api-reference/widgets/infinite-hits/react/#widget-param-cache

```jsx
import { createInfiniteHitsSessionStorageCache } from 'instantsearch.js/es/lib/infiniteHitsCache';

const sessionStorageCache = createInfiniteHitsSessionStorageCache();

<InfiniteHits
  // ...
  cache={sessionStorageCache}
/>
```

**Reply by sbbu** [reply]
This seems to cause an infinite loop when I use it with the `useInfiniteHits` hook...

```typescript
import { createInfiniteHitsSessionStorageCache } from "instantsearch.js/es/lib/infiniteHitsCache";

const sessionStorageCache = createInfiniteHitsSessionStorageCache();

const {
  hits,
  currentPageHits,
  isLastPage,
  showMore,
} = useInfiniteHits({
  cache: sessionStorageCache,
});
```

If I comment out `cache: sessionStorageCache` it works just fine

`Error: Too many re-renders. React limits the number of renders to prevent an infinite loop.`

**Reply by Haroenv** [reply]
are you creating the cache in render?

**Reply by sbbu** [reply]
Yes, that was the problem.. Thank you!

**Reply by nandirb** [reply]
@sbbu How did you solve the problem? I face to same issue when using createInfiniteHitsSessionStorageCache

**Reply by Haroenv** [reply]
@nandirb are _you_ calling the function inside your render as well?

**Reply by nandirb** [reply]
Yes @sbbu
